File: project/TempCode/LSHMinHash.java
public class LSHMinHash extends LSH {
    private final MinHash mh;

    public LSHMinHash(final int bands, final int buckets, final int object_size, final int permutationNumber) {
        super(bands, buckets);
        this.mh = new MinHash(permutationNumber, object_size);
    }

    // Calculate the signature
    public final int[] getSigs(final int[] object) {
    	return this.mh.signature(object);
    }
    
    // Calculate the LSH hash
    public final int[] hash(final int[] sigs) {
        return hashSignature(sigs);
    }
}

File: project/TL_SWAR_analysis/LSHMinHash.cpp
//
//  LSHMinHash.cpp
//  LLVMDemangle
//
//  Created by chauncey on 2018/4/9.
//

#include "LSHMinHash.hpp"
#include "MinHash.hpp"
using namespace SWAR_TL;

// Calculate the signature
int* LSHMinHash::getSigs(int* object, int object_size){
    return mh->signature(object, object_size);
}

// Calculate the LSH hash
int* LSHMinHash::hash(int* sigs, int sig_size){
    return hashSignature(sigs, sig_size);
}
